CodeEditModules/Modules/Git/src/Client/Models/ChangedFiles.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  ChangedFiles.swift
+//  
+//
+//  Created by Nanashi Li on 2022/05/05.
+//
+
+import Foundation
+import SwiftUI
+import WorkspaceClient
+
+public struct ChangedFiles: Codable, Hashable, Identifiable {
+    public var id = UUID()
+
+    /// Change type is to tell us whether the type is a new file, modified or deleted
+    public let changeType: String
+
+    /// Link of the file
+    public let fileLink: URL
+
+    /// Use it like this
+    /// ```swift
+    /// Image(systemName: item.systemImage)
+    /// ```
+    public var systemImage: String {
+        if isFolder {
+            return "folder.fill"
+        } else {
+            return FileIcon.fileIcon(fileType: fileType)
+        }
+    }
+
+    /// Returns the file name (e.g.: `Package.swift`)
+    public var fileName: String {
```

Comment:
do we need to have a comptued property ? We can retrieve this info directly from the URL

Reply:
Yes because we use it do display on the list.

Reply:
please do not mark comments as resolved if not resolved

Reply:
when we display this on the list we could actually access the URL directly

Reply:
Do you want clean code or do you want a mess because that's what you are suggesting right now.